### ISSUE TITLE
"https://" prefix on push config is required on some systems and not on others

### ISSUE THREAD

**Issue by AaronFriel** [pulumi/pulumi-docker#472]
I found that when using ECR login using the credential tool, I received a confusing error message: "no basic auth credentials".

## Steps to reproduce

1. Log in via `aws ecr get-login-password`.

```
aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin [registry hostname]
```

This created an authconfig entry in docker/config.json for `[registry hostname]` **without** `https://`. 

2. Use a program to construct an ECR Registry and v4 Image

## Expected behavior

Successful push.

## Actual behavior

Error "no basic auth credentials".

## Root causing

The `registryServer` on line 210 here began with `https://`, but the `authConfigs` map contained only hostnames without `https://`:

https://github.com/pulumi/pulumi-docker/blob/f41fbf46ea122225aa6e7c11e36b46cb62614675/provider/image.go#L210-L216

**Comment by mikhailshilkov** [comment]
@AaronFriel Could you please add a `kind/*` label? That helps keep our reporting clean. Thank you!

**Comment by guineveresaenger** [comment]
For ACR and Dockerhub, the `authConfigs` map does need the `https://` prefix for lookup.

A side note: I am fairly certain most users would pass in the registry username and password/token-as-password directly from creating the registry inline in the program.

**Comment by guineveresaenger** [comment]
Update:

This appears to be a subtle difference in the credential helpers used to call `GetAllCredentials` between either OSes or docker engine versions.

Solution to move forward: Try both lookups in `authConfig`.

**Comment by guineveresaenger** [comment]
Fixed in #480
